src/storage/v2/indices/text_index.cpp

```diff
@@ -191,19 +234,22 @@ void TextIndex::CreateIndex(const std::string &index_name, LabelId label, memgra
 
     auto vertex_properties = v.Properties(View::NEW).GetValue();
     LoadNodeToTextIndices(v.Gid().AsInt(), SerializeProperties(vertex_properties, db),
-                          {&index_.at(index_name).context_});
+                          StringifyProperties(vertex_properties), {&index_.at(index_name).context_});
   }
 
   CommitLoadedNodes(index_.at(index_name).context_);
 }
 
-void TextIndex::RecoverIndex(const std::string &index_name, LabelId label,
+void TextIndex::RecoverIndex(const std::filesystem::path &storage_dir, const std::string &index_name, LabelId label,
                              memgraph::utils::SkipList<Vertex>::Accessor vertices, NameIdMapper *name_id_mapper) {
   if (!flags::AreExperimentsEnabled(flags::Experiments::TEXT_SEARCH)) {
     throw query::TextSearchDisabledException();
   }
 
-  CreateEmptyIndex(index_name, label);
+  // Clear Tantivy-internal files if they exist from previous sessions
+  std::filesystem::remove_all(storage_dir / kTextIndicesDirectory / index_name);
```

Comment:
We persist data with snapshots and WAL files, not with Tantivy’s internal files. This has to do with the fact that we need to load GIDs into Tantivy so that we can pair documents with nodes, and GIDs are not preserved in recovery.